resolver/actions.go

```diff
@@ -73,6 +73,40 @@ func (g *Actions) Resolve(ctx context.Context, value string) (string, error) {
 	return fmt.Sprintf("%s@%s", name, sha), nil
 }
 
+func (g *Actions) Upgrade(ctx context.Context, value string) (string, error) {
+	githubRef, err := ParseActionRef(value)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse github ref: %w", err)
+	}
+	owner := githubRef.owner
+	repo := githubRef.repo
+	path := githubRef.path
+	ref := githubRef.ref
+
+	// Do not upgrade main branch refs.
+	if ref == "main" || ref == "master" {
+		return value, nil
+	}
```

Comment:
I'm not convinced this is a robust enough strategy. Someone could store actions in a branch other than main and reference it. We could check if you are using sem-ver and only upgrade if so - but I couldn't think of a good way of determining that. We also could add a new ratchet comment like `# ratchet:no-upgrade` to indicate it should be skipped for upgrades. We'd have to allow multiple attributes per line comment then.

@sethvargo wdyt?

Reply:
I came up with a strategy for this. At the cost of another query to GH we check if the ref is for a branch. If so then we skip it.